internal/manager/run.go

```diff
@@ -65,6 +67,13 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
 		return fmt.Errorf("unable to start proxy cache server: %w", err)
 	}
 
+	var ConfigReady healthz.Checker = func(_ *http.Request) error {
```

Comment:
isn't Readiness Check ?  proxy could be healthy, but KIC might not be able to connect due to all possible network/configuration reasons
```suggestion
	var ConfigReady proxyConnection.Checker = func(_ *http.Request) error {
```

Reply:
healthz already includes both readiness and liveness, kindly ignore the variable name change.

Reply:
Does Go permit such definitions? Trying to do that gives me a syntax error.

The reason it's binding a new anonymous function to a var is because the function needs to call `IsReady()` on the specific proxy instance returned by `setupProxyServer()` but still needs to satisfy the Checker type (only accepts a request argument). I can either stick the anon function directly into the `mgr.AddReadyzCheck()` call, e.g.

```
diff --git a/internal/manager/run.go b/internal/manager/run.go
index 4d3f4459..85b90094 100644
--- a/internal/manager/run.go
+++ b/internal/manager/run.go
@@ -67,13 +67,6 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
                return fmt.Errorf("unable to start proxy cache server: %w", err)
        }
 
-       var ConfigReady healthz.Checker = func(_ *http.Request) error {
-               if proxy.IsReady() {
-                       return nil
-               }
-               return errors.New("proxy not yet configured")
-       }
-
        setupLog.Info("deploying all enabled controllers")
        controllers, err := setupControllers(mgr, proxy, c)
        if err != nil {
@@ -93,7 +86,12 @@ func Run(ctx context.Context, c *Config, diagnostic util.ConfigDumpDiagnostic) e
        if err := mgr.AddHealthzCheck("health", healthz.Ping); err != nil {
                return fmt.Errorf("unable to setup healthz: %w", err)
        }
-       if err := mgr.AddReadyzCheck("check", ConfigReady); err != nil {
+       if err := mgr.AddReadyzCheck("check", func(_ *http.Request) error {
+               if proxy.IsReady() {
+                       return nil
+               }
+               return errors.New("proxy not yet configured")
+       }); err != nil {
                return fmt.Errorf("unable to setup readyz: %w", err)
        }
```
Which is ugly as all hell, but works, or make a new function that takes a proxy and returns a Checker that uses it. The latter is arguably the same thing, but a bit less ugly.

Reply:
Define a variable as a function results in syntax error ?  does not sound right.
error message ? 

the two implementation is the same, really personal decision.

Reply:
_Not_ assigning to a variable and putting the `func whatever()` within the body of another function is invalid.

70d5256cb341c37fdd41cd1d984212edf5bb41a8 does the inline definition within the `AddReadyzCheck()` call.